README.md

```diff
@@ -60,3 +60,33 @@ New state map matches old one
 # It's finished, so we can now go and rewrite the DB
 $ psql synapse < out.data
 ```
+
+## Using as python library
+
+The compressor can also be built into a python library as it uses PyO3. It can be
+built and installed into the current virtual environment by running `maturin develop`
+
+The following code does exactly the same as the command-line example from above:
+```
+import synapse_compress_state as comp
+
+comp.run_compression(
+  db_url="postgresql://localhost/synapse",
+  room_id="!some_room:example.com",
+  output_file="out.sql",
+  transactions=True
+)
+```
+
+Note: since this library uses Jemalloc, you might get an error of the form:
+```
+ImportError: /[LONG_PATH]/synapse_compress_state.abi3.so: cannot allocate memory in static TLS block
+```
+If this happens then try running the following:
+```
+LD_PATH=/[LONG_PATH]/synapse_compress_state.abi3.so ./my_python_script
+```
+Or just try disabling jemalloc:
+```
+$ maturin develop --cargo-extra-args="--no-default-features"
```

Comment:
I think we should add `cargo-extra-args = "--no-default-features` to a pyproject.toml so that we always do this when building the python bindings.

Reply:
done